vignettes/cookbook_preparatory_steps.Rmd

```diff
@@ -14,6 +14,233 @@ knitr::opts_chunk$set(
+
+### `r2dii.*` suite of R packages
+
+The `r2dii.*` suite of R packages is the main software tool that you will use to run the PACTA for Banks analysis.
+
+::: {.pkgdown-release}
+You can install the `r2dii.*` suite of R packages from any CRAN mirror by running the following command in R:
+
+``` r
+install.packages(c("r2dii.data", "r2dii.match", "r2dii.analysis", "r2dii.plot"))
+```
+:::
+
+::: {.pkgdown-devel}
+You can install the development versions of the `r2dii.*` suite of R packages from GitHub with:
+
+``` r
+# install.packages("pak")
+pak::pak(c("RMI-PACTA/r2dii.data", "RMI-PACTA/r2dii.match", "RMI-PACTA/r2dii.analysis", "RMI-PACTA/r2dii.plot"))
+```
+
+We use the [`pak` package](https://pak.r-lib.org/index.html) as a simple [tool to install packages from GitHub](https://pak.r-lib.org/reference/pak.html).
+:::
+
+#### Connecting to GitHub from RStudio
+
+Note that if you choose to install the `r2dii.*` suite of R packages from GitHub, you will need to have:
+
+(1) registered a [GitHub](https://github.com/) account,
+(2) `git` installed locally,
+(3) set up credentials so that RStudio can communicate with GitHub.
+
+You can find more information on how to do this using the following resources:
+
+* [Happy Git and GitHub for the useR](https://happygitwithr.com/) is a great and comprehensive resource that takes you through the process of setting up git and GitHub with RStudio, including registering a GitHub account, installing git, and connecting RStudio to GitHub.
+* Additional information on managing your GitHub connection from within RStudio can be found in the usethis package documentation, for example on [managing git credentials](https://usethis.r-lib.org/articles/git-credentials.html).
+
+If you only plan to use GitHub to install this package or other packages as shown above, you will not have to have a deep understanding of all the git commands, so there is no need to be overwhelmed by the complexity of git.
+
+
+### Required R packages
+
+The `r2dii.*` suite of R packages depends on a number of other R packages. These dependencies will be installed automatically when you install the `r2dii.*` suite of R packages. The required packages are:
+
+```{r imports, echo = FALSE, results = 'asis'}
+imports <- paste(
+  utils::packageDescription("r2dii.data")[["Imports"]],
+  utils::packageDescription("r2dii.match")[["Imports"]],
+  utils::packageDescription("r2dii.analysis")[["Imports"]],
+  utils::packageDescription("r2dii.plot")[["Imports"]],
+  sep = ", "
+)
+imports <- sort(unique(unlist(strsplit(x = imports, split = ",[[:space:]]"))))
+cat(paste0("`", paste0(imports, collapse = "`, `"), "`"))
+```
+
+### Suggested R packages
+
+The suggested packages are not required to run the analysis, but they are used in the examples and vignettes provided with the package:
```

Comment:
This feels like we may want to clarify that this means that they are being installed automatically, or, if not, that we think it is beneficial to install them. Let's assume the user does not find that obvious

Reply:
I don't think `install.packages()` installs packages in "Suggests" automatically, unless the user has non-default settings on their system somehow

Reply:
in that case I think it would be god to suggest that it is a good idea to install the suggested packages, to ensure all documentation is readily available.

Reply:
I believe if something is installed from CRAN, it's typically/preferably installed as a "binary" which is already compiled including its documentation, and it does not install the "suggested" packages because they are not needed for normal functioning of the package as installed including viewing/using its documentation. I might be somewhat confused about that, but if I'm right, I'd actually suggest removing this section entirely since mentioning the packages in Suggests just unnecessarily complicates things for the user with something they don't really need to know about at all.

Reply:
that sounds good too

Reply:
removed section about packages in "Suggests"